### DISCUSSION TITLE
Speed deprecation?

### DISCUSSION THREAD

**Discussion by Soulreaver90** [vladmandic/sdnext#1310] [General]
Keeping this short. I have a 6700xt and my average it/s at 512x512 using stock settings have always been 6.50 it/s. I've been using Auto's SD for months with the same speed. I had used Vlad's fork a month or so ago, and had also gotten roughly the same speed. After a month of using Auto's SD, I decided to give this fork a shot again especailly with all the new bells and whistles. However, my speeds at stock 512x512 is at 5 it/s! I reinstalled again from scratch, manually reinstalled torch and unchecked a bunch of settings such as live preview, but I am still only getting 5 it/s. Anyone who is up to date on this fork can shed some light on this? Is this a known issue? I don't want to open this as an issue unless others are having the same experience.

**Comment by Aptronymist** [comment]
> Keeping this short. I have a 6700xt and my average it/s at 512x512 using stock settings have always been 6.50 it/s. I've been using Auto's SD for months with the same speed. I had used Vlad's fork a month or so ago, and had also gotten roughly the same speed. After a month of using Auto's SD, I decided to give this fork a shot again especailly with all the new bells and whistles. However, my speeds at stock 512x512 is at 5 it/s! I reinstalled again from scratch, manually reinstalled torch and unchecked a bunch of settings such as live preview, but I am still only getting 5 it/s. Anyone who is up to date on this fork can shed some light on this? Is this a known issue? I don't want to open this as an issue unless others are having the same experience.

Can you give us some details on what your settings are? We don't even know your OS, what memory/attention you have selected, etc. Kinda hard to even begin with "it's slower".

**Reply by Soulreaver90** [reply]
Ubuntu 22.04
Default SD settings, all out of the box.

**Reply by Soulreaver90** [reply]
I didn’t mention specs as this was not an issue submission but just a general question to see if anyone shared the same symptoms. 

Current Auto: 6.5 it/s
Current Vlad (current as of when I submitted this question): 5.0 it/s
Vlad’s from a month or so ago: 6.5 it/a

I haven’t tried reinstalling it or trying an old commit. Again all out of the box, default extensions and settings.

**Reply by vladmandic** [reply]
lets try `--experimental` command line flag as well :)

**Reply by Soulreaver90** [reply]
> lets try `--experimental` command line flag as well :)

No difference, caps at 5 it/s. Ill play around with the different memory attention settings.

**Reply by vladmandic** [reply]
also check your precision - fp32/fp16, etc.

**Reply by Soulreaver90** [reply]
it seems to be related to an issue I had with this repo way back. I now remember in the old commits, my stock speed would go from under 1 it/s to 6.5 it/s, however something has changed (for the better I guess), and the speed starts at 5 it/s and goes to 6.5 when I tinker with it. I'll have to go one by one and see what I am doing that "fixes" this issue.

**Reply by vladmandic** [reply]
when you do, let us know

**Reply by Soulreaver90** [reply]
Okay so simple fix. 
Th install defaults "Cross-attention optimization method" to sub-quadratic. Changing this to Doggettx, closing out SD entirely and relaunching it restores my default speeds to 6.5.

**Reply by monoinyo** [reply]
So what

> Okay so simple fix. Th install defaults "Cross-attention optimization method" to sub-quadratic. Changing this to Doggettx, closing out SD entirely and relaunching it restores my default speeds to 6.5.

Interesting... So what do you think Vlad, should we be switching to this dog text optimization method or is there a method to the madness with the sub-quadradic?  Let's assume we have different video cards/setups.

(Really appreciate the work Vlad I hope f1 was a damn blast)

**Reply by mariodian** [reply]
Just FYI, Doggettx crashes the app when upscaling on my Mac and [I may not be the only one](https://github.com/AUTOMATIC1111/stable-diffusion-webui/issues/10122) (both in Automatic1111 and Vlad's UI).

InvokeAI albeit tiny bit slower seems more stable.

**Reply by vladmandic** [reply]
i primarily test on nvidia and which settings are optimal for other platforms comes from community.
if there is consensus, i have no problem changing defaults.
note that last change was relatively recent: <https://github.com/vladmandic/automatic/discussions/99#discussioncomment-6016821>

**Reply by Soulreaver90** [reply]
I might just edit the title to focus on memory attention. I haven’t gotten a chance to do more a-b testing since my last post. I am curious to see if changing to sub-q on Auto’s will give the same results, I don’t recall ever seeing my speeds tank that much but I don’t really stray away from what works. Hope to have more results later. Maybe a poll for amd users one which attention works best for them, no idea.

**Reply by mariodian** [reply]
BTW I set the cross attention back to doggettx on my mac and it's not crashing during upscale anymore. One of the updates within the last week or so had to fix it.

**Comment by Rogal80** [comment]
same here on the last update - speeds go down at least 2times slower 

![image](https://github.com/vladmandic/automatic/assets/112553453/f2ea73aa-35ec-4b2f-bcef-51368f49d86b)

**Reply by vladmandic** [reply]
posts with "same here" are pretty much *never* a good idea. i don't even know what gpu you have.

**Comment by Rogal80** [comment]
my gpu - same here -i meant that yesterday my speed was 2times faster after update is slower on base pass and high res on same settings 
![image](https://github.com/vladmandic/automatic/assets/112553453/acfdabfe-c39e-4c47-9ef2-f59e636f4a60)

**Reply by vladmandic** [reply]
which means its a completely different issue and most likely related to #1285

**Reply by Rogal80** [reply]
actually i reinstall my drivers 2 days ago for this version 
![image](https://github.com/vladmandic/automatic/assets/112553453/8b7ed241-ecd7-4a47-bf0d-cb29cefeee8c)

**Reply by Rogal80** [reply]
but yesterday for example on img 704x384 speed was around 6/it with same loras, without lora and embedings first pass around 8-9 it  - today after update i get 2.5/it - live preview set on full vae

**Reply by vladmandic** [reply]
again, totally different issue than original poster. and there were no changes in core processing in the last few days at all.

**Reply by Rogal80** [reply]
One thing that I did today I just install this git https://github.com/bmaltais/kohya_ss - I am just an artist :) sorry Vlad maybe this git is not related, I'll play with settings and eventually reinstall the repository

**Comment by Disty0** [comment]
Had a similar issue too. 
My case was the live preview. 

Live preview was set to nothing and it was ignoring what i was setting it to.
It was falling back to the Full VAE preview and Full VAE has massive hit on performance.
(768x768; Approximate NN 2.8 it/s, Full VAE 2 it/s.)

Pushed a commit (https://github.com/vladmandic/automatic/commit/eaeb5453b878cfaad9b46cfd73d9719ef45cfd0d) for this and It's now using the preview I set instead of ignoring it.

**Reply by monoinyo** [reply]
> Had a similar issue too. My case was the live preview.
> 
> Live preview was set to nothing and it was ignoring what i was setting it to. It was falling back to the Full VAE preview and Full VAE has massive hit on performance. (768x768; Approximate NN 2.8 it/s, Full VAE 2 it/s.)
> 
> Pushed a commit ([eaeb545](https://github.com/vladmandic/automatic/commit/eaeb5453b878cfaad9b46cfd73d9719ef45cfd0d)) for this and It's now using the preview I set instead of ignoring it.

Changing the live preview helped me too, I forgot I changed it to TAESD (don't even know what it is no idea why I changed it).  That messed with my VAE it seems.  3070 8gb here

**Reply by Soulreaver90** [reply]
Curious to test that out
